cmd/gaia/init/gentx.go

```diff
@@ -80,26 +84,33 @@ following delegation and commission default parameters:
 			}
 			// Run gaiad tx create-validator
 			prepareFlagsForTxCreateValidator(config, nodeID, ip, genDoc.ChainID, valPubKey)
-			createValidatorCmd := cli.GetCmdCreateValidator(cdc)
+			cliCtx, txBldr, msg, err := cli.BuildCreateValidatorMsg(
+				context.NewCLIContext().WithCodec(cdc),
+				authtxb.NewTxBuilderFromCLI().WithCodec(cdc),
+			)
+			if err != nil {
+				return err
+			}
 
 			w, err := ioutil.TempFile("", "gentx")
 			if err != nil {
 				return err
 			}
 			unsignedGenTxFilename := w.Name()
 			defer os.Remove(unsignedGenTxFilename)
-			os.Stdout = w
-			if err = createValidatorCmd.RunE(nil, args); err != nil {
+
+			if err := utils.PrintUnsignedStdTx(w, txBldr, cliCtx, []sdk.Msg{msg}, true); err != nil {
 				return err
 			}
-			w.Close()
 
 			prepareFlagsForTxSign()
 			signCmd := authcmd.GetSignCommand(cdc)
-			if w, err = prepareOutputFile(config.RootDir, nodeID); err != nil {
+
+			outputDocument, err := makeOutputFilepath(config.RootDir, nodeID)
+			if err != nil {
 				return err
 			}
-			os.Stdout = w
+			viper.Set("output-document", outputDocument)
```

Comment:
Can we pass this in another way (not Viper global variables) if possible - maybe through the context?

Reply:
opened an issue for that

Reply:
will be addressed in an ensuing PR